main.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    item_select = []
+    my_each { |i| item_select << i if yield(i) }
+    item_select
+  end
+
+  def my_all?(param = nil)
```

Comment:
To fix linter issues, you can use inline disable rules.

```suggestion
 # rubocop:disable  Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
  def my_all?(param = nil)
```